zvariant/src/signature.rs

```diff
@@ -354,24 +354,39 @@ impl<'a> std::ops::Deref for Signature<'a> {
     }
 }
 
-impl<'a, 'b> PartialEq<Signature<'a>> for Signature<'b> {
-    fn eq(&self, other: &Signature<'_>) -> bool {
-        self.as_bytes() == other.as_bytes()
+/// Evaluate equality of two signatures, ignoring outer parentheses.
+impl<'a, 'b> PartialEq<Signature<'b>> for Signature<'a> {
+    fn eq(&self, other: &Signature<'b>) -> bool {
+        match (
+            self.without_outer_parentheses(),
+            other.without_outer_parentheses(),
+        ) {
+            (Some(lhs_without), None) => lhs_without == other.as_bytes(),
+            (None, Some(rhs_without)) => self.as_bytes() == rhs_without,
+            _ => self.as_bytes() == other.as_bytes(),
+        }
     }
 }
 
+// Assumes that `str` is a valid signature string.
 impl<'a> PartialEq<str> for Signature<'a> {
     fn eq(&self, other: &str) -> bool {
-        self.as_bytes() == other.as_bytes()
+        let other = Signature::from_str_unchecked(other);
+        self.eq(&other)
     }
 }
 
+// Assumes that `&str` is a valid signature string.
 impl<'a> PartialEq<&str> for Signature<'a> {
     fn eq(&self, other: &&str) -> bool {
-        self.as_bytes() == other.as_bytes()
+        let other = Signature::from_str_unchecked(other);
+        self.eq(&other)
     }
 }
 
+// `Eq` docs: "the derive strategy requires all fields are Eq, which isn’t always desired.".
```

Comment:
This would be a different logical change and hence it should be split into it's own commit with its own justification.